src/common/hooks/useTitleHydratedPieces/useTitleHydratedPieces.js

```diff
@@ -0,0 +1,99 @@
+import keyBy from 'lodash/keyBy';
+import { useQuery } from 'react-query';
+
+import {
+  useNamespace,
+  useOkapiKy,
+} from '@folio/stripes/core';
+import {
+  HOLDINGS_API,
+  ITEMS_API,
+  LOCATIONS_API,
+  REQUESTS_API,
+  batchFetch,
+  useOrderLine,
+} from '@folio/stripes-acq-components';
+
+import { getHydratedPieces } from '../../utils';
+import { usePieces } from '../usePieces';
+import { useTitle } from '../useTitle';
+
+export const useTitleHydratedPieces = ({ receivingStatus, titleId } = {}) => {
+  const ky = useOkapiKy();
+  const [namespace] = useNamespace('receiving-title-hydrated-pieces');
+
+  const {
+    title,
+    isLoading: isTitleLoading,
+  } = useTitle(titleId);
+
+  const {
+    orderLine,
+    isLoading: isOrderLineLoading,
+  } = useOrderLine(title?.poLineId);
+
+  const {
+    pieces,
+    isLoading: isPiecesLoading,
+  } = usePieces(
+    {
+      searchParams: {
+        query: `titleId=${titleId} and poLineId==${orderLine?.id} and receivingStatus==${receivingStatus}`,
+      },
+    },
+    { enabled: Boolean(titleId && orderLine?.id && receivingStatus) },
+  );
+
+  const isReferenceDataLoading = (
+    isTitleLoading
+    || isOrderLineLoading
+    || isPiecesLoading
+  );
+
+  const { data, isLoading } = useQuery(
```

Comment:
Just a suggestion - for me it looks more readable and clear when an object of arguments is passed to `useQuery`, for example:
```javascript
const { data, isLoading } = useQuery({
  queryKey: [namespace, pieces],
  queryFn: someFn,
  enabled: !isReferenceDataLoading && Boolean(pieces?.length),
});
```

Reply:
Updated